### PR TITLE
Change jerry_get_own_property_descriptor return type

### PR DESCRIPTION
**Pull request by gergocs**
bool -> jerry_value_t

### CODE REVIEW

File: jerry-core/api/jerry.c [hunk 1]
```diff
@@ -3840,7 +3840,7 @@ jerry_get_own_property_descriptor (const jerry_value_t  obj_val, /**< object val
   if (!ecma_is_value_object (obj_val)
       || !ecma_is_value_prop_name (prop_name_val))
   {
-    return false;
+    return jerry_create_boolean (false);
```
Comment (zherczeg) [inline jerry-core/api/jerry.c:3843]: `return ECMA_VALUE_FALSE`

File: jerry-core/api/jerry.c [hunk 2]
```diff
@@ -3852,14 +3852,13 @@ jerry_get_own_property_descriptor (const jerry_value_t  obj_val, /**< object val
 #if JERRY_BUILTIN_PROXY
   if (ECMA_IS_VALUE_ERROR (status))
   {
-    // TODO: Due to Proxies the return value must be changed to jerry_value_t on next release
     jcontext_release_exception ();
```
Comment (zherczeg) [inline jerry-core/api/jerry.c:3855]: `return jerry_throw(status);`

File: jerry-core/api/jerry.c [hunk 3]
```diff
@@ -3902,7 +3901,7 @@ jerry_get_own_property_descriptor (const jerry_value_t  obj_val, /**< object val
     }
   }
 
-  return true;
+  return jerry_create_boolean (true);
```
Comment (zherczeg) [inline jerry-core/api/jerry.c:3904]: ECMA_VALUE_TRUE

File: jerry-core/include/jerryscript-core.h [hunk 4]
```diff
@@ -225,7 +225,7 @@ jerry_value_t jerry_define_own_property (const jerry_value_t obj_val,
                                          const jerry_value_t prop_name_val,
                                          const jerry_property_descriptor_t *prop_desc_p);
 
-bool jerry_get_own_property_descriptor (const jerry_value_t obj_val,
+jerry_value_t jerry_get_own_property_descriptor (const jerry_value_t obj_val,
                                         const jerry_value_t prop_name_val,
```
Comment (zherczeg) [inline jerry-core/include/jerryscript-core.h:229]: Fix indentation here

**Review by zherczeg — CHANGES_REQUESTED** [review]

**Review by zherczeg — APPROVED** [review]
LGTM

File: jerry-core/api/jerry.c [hunk 5]
```diff
@@ -3852,14 +3852,13 @@ jerry_get_own_property_descriptor (const jerry_value_t  obj_val, /**< object val
 #if JERRY_BUILTIN_PROXY
   if (ECMA_IS_VALUE_ERROR (status))
   {
-    // TODO: Due to Proxies the return value must be changed to jerry_value_t on next release
-    jcontext_release_exception ();
+    return jerry_throw(status);
```
Comment (rerobika) [inline jerry-core/api/jerry.c:3855]: Missing space before parenthesis.

File: docs/02.API-REFERENCE.md [hunk 6]
```diff
@@ -8421,7 +8421,7 @@ Construct property descriptor from specified property.
 **Prototype**
 
 ```c
-bool
+jerry_value_t
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:8424]: I just realized you should also add:
`*Changed in version [[NEXT_RELEASE]]*: return value is changed to jerry_value_t`
Before `**Example**`

File: tests/unit-core/test-api-property.c [hunk 7]
```diff
@@ -63,8 +63,8 @@ main (void)
 
   /* Test: get own property descriptor */
   prop_desc = jerry_property_descriptor_create ();
-  bool is_ok = jerry_get_own_property_descriptor (global_obj_val, prop_name, &prop_desc);
-  TEST_ASSERT (is_ok);
+  jerry_value_t is_ok = jerry_get_own_property_descriptor (global_obj_val, prop_name, &prop_desc);
```
Comment (rerobika) [inline tests/unit-core/test-api-property.c:66]: all `is_ok` need to be freed by `jerry_release_value`

**Review by rerobika — CHANGES_REQUESTED** [review]

File: tests/unit-core/test-api-property.c (hunk 7 above)
Comment (gergocs) [reply tests/unit-core/test-api-property.c:66]: Do i need to free at the end or every time after i use it?

Comment (rerobika) [reply tests/unit-core/test-api-property.c:66]: Every time after no longer needed.

**Review by rerobika — APPROVED** [review]

**Pull request merged**